site/docs/configuration.md

```diff
@@ -291,6 +291,45 @@ Currently, this defaults to `"namespace"`, this will change to `"default"` in th
 }
 ```
 
+### `type` (experimental) {#type}
+
+```ts
+"module" | "commonjs";
+```
+
+> If you're just thinking "I want to write code with the native ECMAScript import and export syntax and have my code work for most people", you likely should not use this feature. This feature allows you to build packages that import Node.js ESM only dependencies and therefore only support being imported from ESM in newer versions of Node.js and some bundlers.
+
+The `type` config being set to `"module"` allows you to generate ECMAScript modules(ESM) compatible with Node.js' implementation of ESM (Node.js ESM), most modern bundlers, etc.
```

Comment:
shouldnt this mention `typeModule` and not `type` with a `"module"` value?

Reply:
`"type": "module"` has to be set as well, the `typeModule` experimental flag is mentioned further down.